### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by forksofpower**
Validate documentation for all extensions:
- Rates:
  - [x] preinstall
  - [x] postinstall
  - [x] extensions.yaml
  - [x] generate readme
- Track Label:
  - [x] preinstall
  - [x] postinstall
  - [x] extensions.yaml
  - [x] generate readme
- Validate Address:
  - [x] preinstall
  - [x] postinstall
  - [x] extensions.yaml
  - [x] generate readme
- Purchase Label:
  - [x] preinstall
  - [x] postinstall
  - [x] extensions.yaml
  - [x] generate readme

### CODE REVIEW

File: extensions/rates/POSTINSTALL.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      validateAddress: 'no_validation',
+      shipTo: {
+        name: 'Amanda Miller',
+        phone: '555-555-5555',
+        addressLine1: '525 S Winchester Blvd',
+        cityLocality: 'San Jose',
+        stateProvince: 'CA',
+        postalCode: '95128',
+        countryCode: 'US',
+        addressResidentialIndicator: 'yes',
```
Comment (andyhite) [inline extensions/rates/POSTINSTALL.md:26]: This field isn't required

File: extensions/rates/POSTINSTALL.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+1.  You can now use the [Firebase Admin SDK][admin_sdk] to add a document:
+
+```js
+admin
+  .firestore()
+  .collection('${param:COLLECTION_PATH}')
+  .add({
+    shipment: {
+      validateAddress: 'no_validation',
```
Comment (andyhite) [inline extensions/rates/POSTINSTALL.md:17]: You're missing some important required fields: `carrierId`, `serviceCode`, and `shipDate`.

File: extensions/rates/POSTINSTALL.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      shipTo: {
+        name: 'Amanda Miller',
+        phone: '555-555-5555',
```
Comment (andyhite) [inline extensions/rates/POSTINSTALL.md:20]: `phone` isn't required on the `shipTo`.

File: extensions/rates/POSTINSTALL.md (hunk 2 above)
Comment (andyhite) [reply extensions/rates/POSTINSTALL.md:17]: Ah, actually – I thought this was the `purchaseLabel` extension. So nevermind.

File: extensions/rates/PREINSTALL.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+Use this extension to fetch rates for shipments from documents added to a specified Cloud Firestore collection.
```
Comment (arielple) [inline extensions/rates/PREINSTALL.md:1]: Use this extension to calculate real-time shipping rates across 50+ global carriers from documents added to a specified Cloud Firestore collection. [ShipEngine](https://www.shipengine.com/) retrieves all possible rates to help brands, ecommerce platforms, and 3PLs to make the best choice for every shipment, whether shipping cost, time to delivery, carrier capabilities or other factors are most important.

**Review by arielple — CHANGES_REQUESTED** [review]
We’ll also need to update a few things in the extension.yaml file from the “Partner guidelines” doc that was shared with us which is slightly different from the reference on the firebase docs site. Specifically: the name includes the firebase product and displayname/description could use a little more color.

name: firestore-shipengine-calculate-rates

displayName: Calculate Shipping Rates with ShipEngine
description:  Calculates real-time shipping rates across 60+ carriers using shipment data written to a Cloud Firestore collection.

**Review by arielple — APPROVED** [review]

**Comment by andyhite** [comment]
@forksofpower how close is this to being done?

**Review by andyhite — APPROVED** [review]
Looks good. I'm going to merge this so I can get the extensions published.

**Pull request merged**